contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,561 @@
+import { IWETH } from "../interfaces/IWETH.sol";
+import { PreciseUnitMath } from "../lib/PreciseUnitMath.sol";
+
+
+contract ExchangeIssuanceZeroEx is Ownable, ReentrancyGuard {
+
+    using Address for address payable;
+    using SafeMath for uint256;
+    using PreciseUnitMath for uint256;
+    using SafeERC20 for IERC20;
+    using SafeERC20 for ISetToken;
+
+    struct IssuanceModuleData {
+        bool isAllowed;
+        bool isDebtIssuanceModule;
+    }
+
+    /* ============ Constants ============== */
+
+    address constant public ETH_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    /* ============ State Variables ============ */
+
+    address public immutable WETH;
+    mapping(address => IssuanceModuleData) public allowedIssuanceModules;
+
+    IController public immutable setController;
+
+    address public swapTarget;
+
+    /* ============ Events ============ */
+
+    event ExchangeIssue(
+        address indexed _recipient,     // The recipient address of the issued SetTokens
+        ISetToken indexed _setToken,    // The issued SetToken
+        IERC20 indexed _inputToken,     // The address of the input asset(ERC20/ETH) used to issue the SetTokens
+        uint256 _amountInputToken,      // The amount of input tokens used for issuance
+        uint256 _amountSetIssued        // The amount of SetTokens received by the recipient
+    );
+
+    event ExchangeRedeem(
+        address indexed _recipient,     // The recipient address which redeemed the SetTokens
+        ISetToken indexed _setToken,    // The redeemed SetToken
+        IERC20 indexed _outputToken,    // The address of output asset(ERC20/ETH) received by the recipient
+        uint256 _amountSetRedeemed,     // The amount of SetTokens redeemed for output tokens
+        uint256 _amountOutputToken      // The amount of output tokens received by the recipient
+    );
+
+    /* ============ Modifiers ============ */
+
+    modifier isWhitelistedIssuanceModule(address _issuanceModule) {
+        require(allowedIssuanceModules[_issuanceModule].isAllowed, "ExchangeIssuance: INVALID ISSUANCE MODULE");
+         _;
+    }
+
+    constructor(
+        address _weth,
+        IController _setController,
+        address[] memory _issuanceModuleAddresses,
+        bool[] memory _issuanceModuleDebtIssuanceFlags,
+        address _swapTarget
+    )
+        public
+    {
+        require(_issuanceModuleAddresses.length == _issuanceModuleDebtIssuanceFlags.length, "ExchangeIssuance: ISSUANCE MODULE ADDRESSES / TYPE FLAGS LENGTH MISMATCH");
+        setController = _setController;
+
+        WETH = _weth;
+        swapTarget = _swapTarget;
+
+        for (uint256 i = 0; i < _issuanceModuleAddresses.length; i++) {
+            _addIssuanceModule(_issuanceModuleAddresses[i], _issuanceModuleDebtIssuanceFlags[i]);
+        }
+
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * Withdraw slippage to selected address
```

Comment:
+ What is the use case for this method? 
+ How does the contract get into a state where tokens / ETH need to be swept from it?

Reply:
For ERC20 tokens it can happen if the 0x swaps somehow return a higher amount of component tokens than is required to issue the set. Also of course someone could mistakenly send ERC20 tokens to this contract address.

For ETH actually I don't see any obvious way right now for it to get stuck in the contract.

Reply:
@cgewecke I assume this answers your question. If not feel free to reopen the thread.